src/Symfony/Component/Messenger/Bridge/Amqp/Transport/Connection.php

```diff
@@ -324,6 +326,10 @@ private function publishOnExchange(\AMQPExchange $exchange, string $body, string
             $amqpStamp ? $amqpStamp->getFlags() : AMQP_NOPARAM,
             $attributes
         );
+
+        if (isset($this->connectionOptions['confirm_timeout'])) {
+            $this->channel()->waitForConfirm($this->connectionOptions['confirm_timeout']);
```

Comment:
I don't think this works. Do you not need to call `\AMQPChannel::confirmSelect` first for this to have any effect?

Reply:
Thanks! I miss it. Now added & double checked by running integration tests related to Amqp.